plugins/out_azure_kusto/azure_kusto.c

```diff
@@ -220,19 +431,639 @@ flb_sds_t execute_ingest_csl_command(struct flb_azure_kusto *ctx, const char *cs
+ * Ingest all data chunks from the file storage streams into Azure Kusto.
+ *
+ * This function iterates over all file storage streams associated with the
+ * given Azure Kusto context. For each
+ * file in the stream, it checks if the file (chunk) is locked or has exceeded
+ * the maximum number of retry attempts. If the chunk is eligible for processing,
+ * it constructs a request buffer from the chunk data, optionally compresses
+ * the payload, and attempts to ingest it into Azure Kusto.
+ *
+ * The function performs the following steps:
+ * 1. Iterate over each file storage stream in the context.
+ * 2. For each file in the stream, check if it is locked or has exceeded
+ *    the maximum retry attempts. If so, skip processing.
+ * 3. Construct a request buffer from the chunk data.
+ * 4. Create a payload from the buffer and optionally compress it if
+ *    compression is enabled.
+ * 5. Load the necessary ingestion resources for Azure Kusto.
+ * 6. Attempt to ingest the payload into Azure Kusto using queued ingestion.
+ * 7. If ingestion is successful, clean up the local buffer file.
+ * 8. Handle errors by unlocking the chunk, incrementing failure counts,
+ *    and logging appropriate error messages.
+ *
+ * @param ctx    Pointer to the Azure Kusto context containing configuration
+ *               and state information.
+ * @param config Pointer to the Fluent Bit configuration structure.
+ *
+ * @return 0 on success, or -1 on failure.
+ */
+static int ingest_all_chunks(struct flb_azure_kusto *ctx, struct flb_config *config)
+{
+    struct azure_kusto_file *chunk;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct mk_list *f_head;
+    struct flb_fstore_file *fsf;
+    struct flb_fstore_stream *fs_stream;
+    flb_sds_t payload = NULL;
+    void *final_payload = NULL;
+    size_t final_payload_size = 0;
+    char *buffer = NULL;
+    size_t buffer_size;
+    int ret;
+    int is_compressed = FLB_FALSE;
+    flb_sds_t tag_sds;
+
+    mk_list_foreach(head, &ctx->fs->streams) {
+        fs_stream = mk_list_entry(head, struct flb_fstore_stream, _head);
+        if (fs_stream == ctx->stream_upload) {
+            continue;
+        }
+
+        mk_list_foreach_safe(f_head, tmp, &fs_stream->files) {
+            fsf = mk_list_entry(f_head, struct flb_fstore_file, _head);
+            chunk = fsf->data;
+
+            /* Locked chunks are being processed, skip */
+            if (chunk->locked == FLB_TRUE) {
+                continue;
+            }
+
+            if (chunk->failures >= ctx->scheduler_max_retries) {
+                flb_plg_warn(ctx->ins,
+                             "ingest_all_old_buffer_files :: Chunk for tag %s failed to send %i times, "
+                             "will not retry",
+                             (char *) fsf->meta_buf, ctx->scheduler_max_retries);
+                if (ctx->delete_on_max_upload_error){
+                    azure_kusto_store_file_delete(ctx, chunk);
+                }else{
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions